### PR TITLE
Add non-minified geofire-utils.js build target

### PR DESCRIPTION
**Pull request by puf**
### Description

This branch adds a new build target to distribute a separate version of geofire-utils.js, which can then be used without any dependency on the Realtime Database.

### Code sample
```
import { encodeGeohash, geohashQueries, distance } from './geofire-utils.js';
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,14 +30,16 @@
   "browser": "dist/geofire.min.js",
   "module": "dist/index.esm.js",
   "typings": "dist/index.d.ts",
+  "utils": "dist/geofire-utils.js",
```
Comment (samtstern) [inline package.json:33]: I was thinking this would be a separate package.  Is this a normal use for a "distribution"?

File: package.json [hunk 2]
```diff
@@ -30,14 +30,16 @@
...
   "files": [
     "dist/**",
     "LICENSE",
     "README.md",
     "package.json"
   ],
   "dependencies": {
-    "@firebase/database-types": "0.x.x"
+    "@firebase/database-types": "0.x.x",
+    "@rollup/plugin-commonjs": "^15.0.0"
```
Comment (samtstern) [inline package.json:42]: If you need more `rollup` deps move them to `devDependencies` so they don't actually ship.

**Comment by samtstern** [comment]
@puf I fixed the tests here:
https://github.com/firebase/geofire-js/pull/198

You should merge `master` into your branch.

File: package.json (hunk 2 above)
Comment (puf) [reply package.json]: My mistake: the plugin was already in `devDependencies`, so I rolled back to that version. No sense in updating all dependencies for just this new release target.

File: package.json (hunk 1 above)
Comment (puf) [reply package.json:33]: I was only thinking of pulling the file of a CDN. Do you think we should also allow it to be installed from NPM separately? In that case it'd have to be a new package indeed.

Comment (samtstern) [reply package.json:33]: If we can figure out how to do another package that would be ideal.  That's how I have it set up for Android.  Then it works better for people installing with `npm` and using bundlers, etc.

**Comment by samtstern** [comment]
@puf I ran `git clean -f -d -X` to get rid of all generated files and then ran `npm install && npm run bootstrap && npm run build` and then I have the following files:

**geofire**
```
$ ls -Ral packages/geofire/dist
total 0
drwxr-xr-x   3 samstern  primarygroup    96B Sep 28 10:55 ./
drwxr-xr-x  12 samstern  primarygroup   384B Sep 28 10:55 ../
drwxr-xr-x  12 samstern  primarygroup   384B Sep 28 10:55 geofire/

packages/geofire/dist/geofire:
total 392
drwxr-xr-x  12 samstern  primarygroup   384B Sep 28 10:55 ./
drwxr-xr-x   3 samstern  primarygroup    96B Sep 28 10:55 ../
-rw-r--r--   1 samstern  primarygroup   559B Sep 28 10:56 GeoCallbackRegistration.d.ts
-rw-r--r--   1 samstern  primarygroup   3.4K Sep 28 10:56 GeoFire.d.ts
-rw-r--r--   1 samstern  primarygroup   557B Sep 28 10:56 GeoFireTypes.d.ts
-rw-r--r--   1 samstern  primarygroup   8.0K Sep 28 10:56 GeoQuery.d.ts
-rw-r--r--   1 samstern  primarygroup   973B Sep 28 10:56 databaseUtils.d.ts
-rw-r--r--   1 samstern  primarygroup    12K Sep 28 10:56 geofire.min.js
-rw-r--r--   1 samstern  primarygroup    48K Sep 28 10:56 index.cjs.js
-rw-r--r--   1 samstern  primarygroup   181B Sep 28 10:56 index.d.ts
-rw-r--r--   1 samstern  primarygroup    48K Sep 28 10:56 index.esm.js
-rw-r--r--   1 samstern  primarygroup    50K Sep 28 10:56 index.js
```

**geofire-common**
```
$ ls -Ral packages/geofire-common/dist
total 0
drwxr-xr-x   3 samstern  primarygroup    96B Sep 28 10:55 ./
drwxr-xr-x  12 samstern  primarygroup   384B Sep 28 10:55 ../
drwxr-xr-x   7 samstern  primarygroup   224B Sep 28 10:55 geofire/

packages/geofire-common/dist/geofire:
total 120
drwxr-xr-x  7 samstern  primarygroup   224B Sep 28 10:55 ./
drwxr-xr-x  3 samstern  primarygroup    96B Sep 28 10:55 ../
-rw-r--r--  1 samstern  primarygroup   4.0K Sep 28 10:55 geofire.min.js
-rw-r--r--  1 samstern  primarygroup    15K Sep 28 10:55 index.cjs.js
-rw-r--r--  1 samstern  primarygroup   5.3K Sep 28 10:55 index.d.ts
-rw-r--r--  1 samstern  primarygroup    14K Sep 28 10:55 index.esm.js
-rw-r--r--  1 samstern  primarygroup    16K Sep 28 10:55 index.js
```

I can clearly see the utilities you implemented in `packages/geofire-common/dist/geofire/index.js` ... is this not working for you?

That said I do think the `rollup` config in `common` needs some cleanup, right now it's 100% cribbed from `geofire` but it needs to be slightly tweaked.

File: packages/geofire-common/rollup.config.js [hunk 3]
```diff
@@ -5,6 +5,7 @@ import { uglify } from 'rollup-plugin-uglify';
 import pkg from './package.json';
 
 const GLOBAL_NAME = 'geofire';
+const UTILS_NAME = 'geofire-utils';
```
Comment (samtstern) [inline packages/geofire-common/rollup.config.js:8]:
```suggestion
const GLOBAL_NAME = 'geofire-common';
```

File: packages/geofire-common/rollup.config.js [hunk 4]
```diff
@@ -39,7 +40,7 @@ const completeBuilds = [{
   {
     input: 'src/index.ts',
     output: {
-      file: 'dist/geofire.js',
+      file: pkg.index,
```
Comment (samtstern) [inline packages/geofire-common/rollup.config.js:43]:
```suggestion
      file: 'dist/geofire-common.js',
```

File: packages/geofire/rollup.config.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
+import commonjs from 'rollup-plugin-commonjs';
+import resolveModule from 'rollup-plugin-node-resolve';
+import typescript from 'rollup-plugin-typescript2';
+import { uglify } from 'rollup-plugin-uglify';
+import pkg from './package.json';
+
+const GLOBAL_NAME = 'geofire';
+const UTILS_NAME = 'geofire-utils';
```
Comment (samtstern) [inline packages/geofire/rollup.config.js:8]:
```suggestion
const GLOBAL_NAME = 'geofire';
```

File: packages/geofire/rollup.config.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+      file: pkg.browser,
+      format: 'umd',
+      name: GLOBAL_NAME
+    },
+    plugins: [...plugins, uglify()]
+  },
+  {
+    input: 'src/index.ts',
+    output: {
+      file: pkg.index,
```
Comment (samtstern) [inline packages/geofire/rollup.config.js:43]:
```suggestion
      file: 'dist/geofire.js',
```

**Comment by puf** [comment]
> @puf I ran `git clean -f -d -X` to get rid of all generated files and then ran `npm install && npm run bootstrap && npm run build` and then I have the following files:
...

I end up with the same. Not sure what was different about it last week, but the cleanup indeed might have helped.

Should the `packages/geofire-common/dist/geofire` actually be `packages/geofire-common/dist/geofire-common` to so we can push it to npm as its own package?

**Comment by puf** [comment]
LGTM

We can improve coverage later, and will add tasks for strengthening the types.

**Pull request merged**